### PR TITLE
perf: (pool): Refactor pool staking limit calls

### PR DESCRIPTION
**Pull request by memoyil**
Uses multicall instead of call per pool.

With farm call refactor, total pools calls will be down from ~900 to ~10.

To review:

https://deploy-preview-2860--pancakeswap-dev.netlify.app/

### CODE REVIEW

**Review by chef-jojo — APPROVED** [review]
LGTM

File: src/state/pools/fetchPools.ts [hunk 1]
```diff
@@ -102,11 +103,16 @@ export const fetchPoolsStakingLimits = async (
     .filter((p) => !poolsWithStakingLimit.includes(p.sousId))
 
   // Get the staking limit for each valid pool
-  // Note: We cannot batch the calls via multicall because V1 pools do not have "poolLimitPerUser" and will throw an error
-  const stakingLimitPromises = validPools.map((validPool) => fetchPoolStakingLimit(validPool.sousId))
-  const stakingLimits = await Promise.all(stakingLimitPromises)
-
-  return stakingLimits.reduce((accum, stakingLimit, index) => {
+  const poolStakingCalls = validPools.map((validPool) => {
+    const config = poolsConfig.find((pool) => pool.sousId === validPool.sousId)
```
Comment (chef-jojo) [inline src/state/pools/fetchPools.ts:107]: `validPools` is filtered from `poolsConfig`. Why do we need to `poolsConfig.find` again?

Comment (memoyil) [reply src/state/pools/fetchPools.ts]: Done

**Comment by chef-jojo** [comment]
@memoyil  sorry I forget to mention, we have a new standard that will utilizing the `hasUserLimit` check https://github.com/pancakeswap/pancake-frontend/pull/3002 `hasUserLimit` maybe false even we have `poolLimitPerUser` 

We'll need to make each multcall with [hasUserLimit, poolLimitPerUser] 😅

**Comment by memoyil** [comment]
> @memoyil sorry I forget to mention, we have a new standard that will utilizing the `hasUserLimit` check #3002 `hasUserLimit` maybe false even we have `poolLimitPerUser`
> 
> We'll need to make each multcall with [hasUserLimit, poolLimitPerUser] 😅

Done 👍

**Review by chef-jojo — APPROVED** [review]
👍

**Pull request merged**